docs/txp.md

```diff
@@ -0,0 +1,486 @@
+is verified according to new GState. If transaction is considered
+invalid, all temporary modifications are discarded. If all
+transactions are valid, temporary modifications are committed.
+
+Let's assume that version comparison as described at the end of previos section (Unknown data handling) is done.
+Let's denote flag `verifyAllIsKnown` and assign it value:
+ * `True` if check is to be done
+    - I.e. only known data should be considered valid (unknown data is prohibitied)
+ * `False` otherwise
+
+#### General checks
+
+Checks, that are considering whole transaction, not inputs/outputs in particular.
+
+- *Consistency check*: number of inputs in a transaction must be the same as number of witnesses for this transaction
+- *Attributes known check*: if `verifyAllIsKnown` is `True`, all transaction attributes must be known.
+- *Transaction size check*: size of transaction doesn't exceed size limit
+    - Transaction size is computed as number of bytes in serialized `TxAux` (which contains transaction and its witness)
+    - Transaction size limit is part of `BlockVersionData`.
+    - If the size of transaction is greater than the limit from the adopted `BlockVersionData`, transaction is
+invalid.
+- *Bootstrap era check*: if the block was created during bootstrap era, all
+output addresses must have `BootstrapEraDistr` distribution
+  - Predicate "was created during bootstrap era" is checked with block's slot field
+    and adopted `BlockVersionData` (which contains data which slot is defined last for bootstrap era)
+
+Also there is a check which applies to whole `TxPayload`:
+
+- Number of transactions must be the same as number of `TxWitness`es.
+
+#### Outputs checks
+
+For each output address we perform the following checks:
+1. If `verifyAllIsKnown` is `True`, address' attributes must be known.
+2. If `verifyAllIsKnown` is `True`, address' type must be known.
+3. Address can't be a redeem address.
+
+#### Inputs checks
+
+For each input `i` we perform:
+
+- *Inputs known check*: if `verifyAllIsKnown` is `True`, type of input
+  `i` should have known type
+- *UTXO check*: If input `i` is of type `TxInUtxo`, it should have corresponding record in UTXO
+- *Difference check*: input `i` must be different from all other
+  inputs in transaction. This check is performed only if all inputs
+  have known type.
+
+In addition we perform number of consolidated checks (checks which consider whole set of inputs):
+
+* Witness check (separately for each witness)
+* Sums check (check inputs sum is greater than or equal to outputs sum)
+* Transaction fee check (check transaction fee is fine)
+
+##### Witness checks
+
+Witness check for any type is two-fold:
+
+* check that witness corresponds to the address of corresponding input
+* check witness is correct
+
+Witness checks are omitted if (and only if):
+* There is an unknown transaction input.
+
+We describe checks for each type in folowing sections.
+
+###### Public key witness
```

Comment:
>`######`

omg, I am sad it's maximal possible depth of `#`
tbh I don't like so deep subsections, the indent doesn't differ from outer subsection 
also, I would write which particular witness you describe, `TxInWitness` in this case.

Reply:
>  I would write which particular witness you describe, TxInWitness in this case.
Elaborate please

Reply:
I meant we also have `TxWitness`

Reply:
Where would you write it?